client/ayon_substancepainter/plugins/publish/extract_map_as_single_output.py

```diff
@@ -0,0 +1,96 @@
+import clique
+import os
+
+from ayon_core.pipeline import publish
+from ayon_core.lib import (
+    get_oiio_tool_args,
+    run_subprocess,
+)
+
+
+def convert_texture_maps_for_udim_export(staging_dir, image_outputs, has_udim=False):
+    if has_udim:
+        collections, remainder = clique.assemble(image_outputs, minimum_items=1)
+        return [
+            os.path.join(
+                staging_dir,
+                collection.format(pattern="{head}{padding}{tail}")
+            )
+            for collection in collections
+        ]
+    else:
+        return [
+            os.path.join(staging_dir, output) for output in image_outputs
+        ]
+
+
+def convert_texture_maps_as_single_output(staging_dir, source_image_outputs,
+                                          dest_image_outputs, has_udim=False,
+                                          log=None):
+    oiio_tool_args = get_oiio_tool_args("oiiotool")
+
+    source_maps = convert_texture_maps_for_udim_export(
+        staging_dir, source_image_outputs, has_udim=has_udim)
+    dest_map = next(convert_texture_maps_for_udim_export(
+        staging_dir, dest_image_outputs, has_udim=has_udim
+        ), None)
+
+    log.info(f"{source_maps} composited as {dest_map}")
+    oiio_cmd = oiio_tool_args + source_maps + [
+        "--over", "-o",
+       dest_map
+    ]
+
+    subprocess_args = " ".join(oiio_cmd)
+
+    env = os.environ.copy()
+    env.pop("OCIO", None)
+    log.info(" ".join(subprocess_args))
+    try:
+        run_subprocess(subprocess_args, env=env)
+    except Exception:
+        log.error("Texture maketx conversion failed", exc_info=True)
+        raise
```

Comment:
We should not be joining this ourselves and pass the joined command to run_subprocess but instead pass it the list of arguments.

```suggestion
    env = os.environ.copy()
    env.pop("OCIO", None)
    log.info(" ".join(oiio_cmd))
    try:
        run_subprocess(oiio_cmd, env=env)
    except Exception as exc:
        raise RuntimError("Flattening texture stack to single output image failed") from exc
```

Also, please add a todo as to WHY we need to remove `OCIO` var? What does it break if it remains specified?